Objective-C/Internal/CBLFLDict.h

```diff
@@ -0,0 +1,30 @@
+//
+//  CBLFLDict.h
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 4/23/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "Fleece+CoreFoundation.h"
+@class CBLC4Document;
+@class CBLDatabase;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface CBLFLDict : NSObject
```

Comment:
I need more time to work on this as I don't fully understand about ARC on C++. This shouldn't change the way the other platform will be porting the code as well.

https://github.com/couchbase/couchbase-lite-ios/issues/1748